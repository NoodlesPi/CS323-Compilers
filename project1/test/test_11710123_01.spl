int main() {
    int a = 0x123, b = 123;
    char c = 'c';
    float d = .1, e = 1.;
    return 0;
}