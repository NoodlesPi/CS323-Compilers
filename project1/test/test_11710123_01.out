Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: main
        LP
        RP
      CompSt (1)
        LC
        DefList (2)
          Def (2)
            Specifier (2)
              TYPE: int
            DecList (2)
              Dec (2)
                VarDec (2)
                  ID: a
                ASSIGN
                Exp (2)
                  INT: 291
              COMMA
              DecList (2)
                Dec (2)
                  VarDec (2)
                    ID: b
                  ASSIGN
                  Exp (2)
                    INT: 123
            SEMI
          DefList (3)
            Def (3)
              Specifier (3)
                TYPE: char
              DecList (3)
                Dec (3)
                  VarDec (3)
                    ID: c
                  ASSIGN
                  Exp (3)
                    CHAR: 'c'
              SEMI
            DefList (4)
              Def (4)
                Specifier (4)
                  TYPE: float
                DecList (4)
                  Dec (4)
                    VarDec (4)
                      ID: d
                    ASSIGN
                    Exp (4)
                      FLOAT: 0.100000
                  COMMA
                  DecList (4)
                    Dec (4)
                      VarDec (4)
                        ID: e
                      ASSIGN
                      Exp (4)
                        FLOAT: 1.000000
                SEMI
        StmtList (5)
          Stmt (5)
            RETURN
            Exp (5)
              INT: 0
            SEMI
        RC
